### ISSUE TITLE
Staging Review finding: Date errors not announced to screen readers on focus

### ISSUE THREAD

**Issue by jasonday** [department-of-veterans-affairs/va.gov-team#125330]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** Benefits Intake Optimization (Aquia)
**Product:** Form 21-4192 and Form 21-2680
**Feature:** MVP Form Digitization

## Findings details
**VA.gov Experience Standard - issue:** [User can't identify an element or its state.](https://design.va.gov/about/experience-standards/#user-can-identify-an-element-and-its-state)
**VA.gov Experience Standard - category:** [Usability](https://design.va.gov/about/experience-standards/#category-usability)
**Launch-blocking:** No (due to design system issue)
**Design System review:** Yes
**Collab Cycle Reviewer:** @jasonday (Accessibility)

### Description

When there is an error with a date component, the error is not announced when any of the date fields are focused. Additionally, the fields are not identified as being `invalid` due to `aria-invalid="false"` on a field in error.  As a result, screen reader users may not be able to determine that the field is in error and may not be able to proceed. 

This is especially problematic when the date field group is not the first element in error. In this scenario, the date error is not announced to screen readers, and the screen reader user receives no additional communication of the error when navigating through the experience. 

This applies to all date components within 21-492 and 21-2680.

Note: The severity of this issue is critical, but due to the fix likely residing with the Design System Team, it is not launch-blocking (at this time). @humancompanion-usds 

<details><summary>Link, screenshot or steps to recreate</summary>

- using Jaws in Chrome
- Navigate to step 1 of 7 Veteran information
- Don't enter any information
- TAB past all fields and click Continue
- Focus moves to the first field group in error
- TAB to the fields in the date component
- Note screen reader announcement
  - Screen reader does not announce the error message, nor is the field indicated as invalid

<img width="376" height="414" alt="date error" src="https://github.com/user-attachments/assets/9351f21c-b730-49a4-b789-06065b5040fb" />

</details>

## Recommended action

Ensure that the date fields are identified as in error programmatically using `aria-invalid="true"` and ensure that the error message is associated with the fields in error, so that screen readers announce the error when the fields have focus. 

Note:
- When the parent group has an error message, the fields within the group have `aria-invalid="false"` on them.  This is a mismatch in experience, so screen reader users may not be able to proceed, as the date error is not perceivable to them.

`<select class="usa-select" aria-describedby="input-message" aria-invalid="false" id="options" name="root_veteranInformation_dateOfBirthMonth" required="" part="select">...</select>`

## References

- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)
- **WCAG Success Criteria:** [WCAG 1.3.1 Info and Relationships - Level A](https://www.w3.org/TR/WCAG22/#info-and-relationships)
- **Modality:** Screen reader
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

_How did this touchpoint go? [Give feedback on the Collaboration Cycle](https://dj540s05.optimalworkshop.com/questions/f54355271969f160ef41182ba9ceb61c) at any time._
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue; or if you believe the finding should not be fixed; comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by LinnJS** [comment]
> Work on this will be done in [Forms failing to announce errors on submit (Phase 2: date components)](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5208)

@caw310 Looks like this work will also fix this [other similar ticket for 0779](https://github.com/department-of-veterans-affairs/va.gov-team/issues/125338) right?

**Comment by leahkeeler** [comment]
Platform Bug - awaiting fix

**Comment by caw310** [comment]
This work is done. [Forms failing to announce errors on submit (Phase 2: date components)](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/5208) so it is no longer a blocker.  Please confirm.
